internal/service/implement/auth_service.go

```diff
@@ -16,37 +18,66 @@ import (
 )
 
 type AuthService struct {
-	customerRepository repository.CustomerRepository
+	customerRepository       repository.CustomerRepository
 	authenticationRepository repository.AuthenticationRepository
 }
 
 func NewAuthService(customerRepository repository.CustomerRepository, authenticationRepository repository.AuthenticationRepository) service.AuthService {
 	return &AuthService{
-		customerRepository: customerRepository,
+		customerRepository:       customerRepository,
 		authenticationRepository: authenticationRepository,
 	}
 }
 
 func (service *AuthService) Register(ctx *gin.Context, registerRequest model.RegisterRequest) error {
-	err := service.customerRepository.RegisterCommand(ctx, registerRequest)
-	return err
+	existsCustomer, err := service.customerRepository.GetOneByEmailQuery(ctx, registerRequest.Email)
+	if err != nil {
+		return err
+	}
+	if existsCustomer != nil {
+		return errors.New("Email have already registered")
+	}
+	hashPW, err := bcrypt.GenerateFromPassword([]byte(registerRequest.Password), bcrypt.DefaultCost)
+	if err != nil {
+		return err
+	}
```

Comment:
t nghĩ nên tạo 1 cái bcrypt utils thay vì sài thẳng ở đây

Reply:
checked

Reply:
t suggest tạo 1 struct tên là BcryptEncoder, rồi inject cái struct đó vô service này

Reply:
cách tạo thì tạo theo mẫu thằng middleware này 

[https://github.com/21CLC01-WNC-Banking/WNC-Banking-BE/blob/main/internal/controller/http/middleware/verify_token.go](https://github.com/21CLC01-WNC-Banking/WNC-Banking-BE/blob/main/internal/controller/http/middleware/verify_token.go)